src/services/swap_service.ts

```diff
@@ -233,6 +255,57 @@ export class SwapService {
             .filter(p => p) as GetTokenPricesResponse;
         return prices;
     }
+    private async _getSwapQuoteForWethAsync(
+        params: CalculateSwapQuoteParams,
+        isUnwrap: boolean,
+    ): Promise<GetSwapQuoteResponse> {
+        const {
+            from,
+            buyTokenAddress,
+            sellTokenAddress,
+            buyAmount,
+            sellAmount,
+            affiliateAddress,
+            gasPrice: providedGasPrice,
+        } = params;
+        const amount = buyAmount || sellAmount;
+        if (amount === undefined) {
+            throw new Error('sellAmount or buyAmount required');
+        }
+        const data = (isUnwrap
+            ? this._wethContract.withdraw(amount)
+            : this._wethContract.deposit()
+        ).getABIEncodedTransactionData();
+        const value = isUnwrap ? ZERO : amount;
+        const affiliatedData = this._attributeCallData(data, affiliateAddress);
+        const gasPrice = providedGasPrice || (await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync());
+        const gasEstimate = await this._estimateGasOrThrowRevertErrorAsync({
+            to: this._wethContract.address,
+            data: affiliatedData,
+            from,
+            value,
+            gasPrice,
+        });
+        const suggestedGasEstimate = gasEstimate.times(GAS_LIMIT_BUFFER_PERCENTAGE + 1).integerValue();
```

Comment:
I think we could remove the Buffer as it's pretty deterministic. 

We add a buffer because fallback/Kyber/Curve costs can sometimes not be guaranteed